x-pack/plugins/search_playground/public/components/chat.tsx

```diff
@@ -76,6 +79,21 @@ export const Chat = () => {
     [messages]
   );
 
+  const regenerateMessages = () => {
+    const data = getValues();
```

Comment:
ideally this should be centralised. First step is to create a function that takes the form values and builds up the data object. This can be used here and in the main chat function.